k8s/tools/deploy-pubsub-gke.sh

```diff
@@ -0,0 +1,278 @@
+
+# Create main PubSub Topic/Subscription
+if [[ "$*" != *--no-pubsub* ]] ; then  
+  echo "Enabling the GCP PubSub  API"
+  gcloud -q services --project $DEVSHELL_PROJECT_ID enable pubsub.googleapis.com
+  echo "Creating PubSub topic $INSTANCE_ID"
+  gcloud pubsub topics create $INSTANCE_ID
+  echo "Creating PubSub subscription $INSTANCE_ID"
+  gcloud pubsub subscriptions create $INSTANCE_ID --topic=$INSTANCE_ID --ack-deadline=600 
+
+  # Create internal PubSub PSQ Topic/Subscription
+  echo "Creating PubSub PSQ Topic $INSTANCE_ID-psq"
+  gcloud pubsub topics create "$INSTANCE_ID-psq"
+  echo "Creating PubSub PSQ subscription $INSTANCE_ID-psq"
+  gcloud pubsub subscriptions create "$INSTANCE_ID-psq" --topic="$INSTANCE_ID-psq" --ack-deadline=600
+else
+  echo "--no-pubsub specified. Using pre-existing PubSub topic/subscription $INSTANCE_ID and PSQ topic/subscription $INSTANCE_ID-psq"
+fi
+
+# Update Turbinia config with PubSub parameters
+echo "Updating $TURBINIA_CONFIG with PubSub config"
+sed -i -e "s/^TASK_MANAGER = .*$/TASK_MANAGER = 'PSQ'/g" $TURBINIA_CONFIG
+sed -i -e "s/^PUBSUB_TOPIC = .*$/PUBSUB_TOPIC = '$INSTANCE_ID'/g" $TURBINIA_CONFIG
+sed -i -e "s/^PSQ_TOPIC = .*$/PSQ_TOPIC = '$INSTANCE_ID-psq'/g" $TURBINIA_CONFIG
+
+# Enable Stackdriver Logging and Stackdriver Traceback
+echo "Enabling Cloud Error Reporting and Logging APIs"
+gcloud -q services --project $DEVSHELL_PROJECT_ID enable clouderrorreporting.googleapis.com
+gcloud -q services --project $DEVSHELL_PROJECT_ID enable logging.googleapis.com
+echo "Updating $TURBINIA_CONFIG to enable Stackdriver Traceback and Logging"
+sed -i -e "s/^STACKDRIVER_LOGGING = .*$/STACKDRIVER_LOGGING = True/g" $TURBINIA_CONFIG
+sed -i -e "s/^STACKDRIVER_TRACEBACK = .*$/STACKDRIVER_TRACEBACK = True/g" $TURBINIA_CONFIG
+
+# Enable Prometheus
+echo "Updating $TURBINIA_CONFIG to enable Prometheus application metrics"
+sed -i -e "s/^PROMETHEUS_ENABLED = .*$/PROMETHEUS_ENABLED = True/g" $TURBINIA_CONFIG
+
+# Disable some jobs
+echo "Updating $TURBINIA_CONFIG with disabled jobs"
+sed -i -e "s/^DISABLED_JOBS = .*$/DISABLED_JOBS = $DISABLED_JOBS/g" $TURBINIA_CONFIG
+
+# Set appropriate docker image in deployment file if user specified
+if [[ ! -z "$TURBINIA_SERVER_IMAGE" && ! -z "$TURBINIA_WORKER_IMAGE" ]] ; then
+  echo "Updating deployment files with docker image $TURBINIA_SERVER_IMAGE and $TURBINIA_WORKER_IMAGE"
+  sed -i -e "s/us-docker.pkg.dev\/osdfir-registry\/turbinia\/release\/turbinia-server:latest$/$TURBINIA_SERVER_IMAGE/g" turbinia-server.yaml
+  sed -i -e "s/us-docker.pkg.dev\/osdfir-registry\/turbinia\/release\/turbinia-worker:latest$/$TURBINIA_WORKER_IMAGE/g" turbinia-worker.yaml
+fi
+
+# Deploy to cluster
+echo "Deploying Turbinia to $CLUSTER_NAME cluster"
+./setup-pubsub.sh $TURBINIA_CONFIG
+
+# Make a copy of Turbinia config in user home directory
+echo "Creating a copy of Turbinia config in $HOME/.turbiniarc"
+cp $TURBINIA_CONFIG $HOME/.turbiniarc
```

Comment:
I think in the current `deploy.sh` we create a backup of this file if it exists first.  Could we do that here too instead of just overwriting it?

Reply:
Done